sdc/str_arr_ext.py

```diff
@@ -1543,3 +1543,14 @@ def _sdc_str_arr_operator_mul_impl(self, other):
         return res_arr
 
     return _sdc_str_arr_operator_mul_impl
+
+
+@lower_builtin(operator.is_, StringArrayType, StringArrayType)
+def sdc_str_arr_operator_is(context, builder, sig, args):
```

Comment:
Looks good.

We would need to implement such method for each new series type (e.g. categorical, datetime, etc.)